optaplanner-core/src/main/java/org/optaplanner/core/impl/phase/AbstractPhase.java

```diff
@@ -127,6 +135,8 @@ public void solvingEnded(SolverScope<Solution_> solverScope) {
 
     @Override
     public void phaseStarted(AbstractPhaseScope<Solution_> phaseScope) {
+        phaseIndex = phaseCounter.getPhasesStarted();
+        phaseCounter.phaseStarted(phaseScope);
```

Comment:
Because phases can now be reused, the phase index is only assigned at phase start.